test/e2e/marketplace_test.go

```diff
@@ -0,0 +1,186 @@
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	testNamespace        = "operator-test"
+)
+
+func TestMarketplace(t *testing.T) {
+	operatorsourceList := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	catalogsourceconfigList := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, operatorsourceList)
+	if err != nil {
+		t.Fatalf("failed to add operatorsource custom resource scheme to framework: %v", err)
+	}
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, catalogsourceconfigList)
+	if err != nil {
+		t.Fatalf("failed to add catalogsourceconfig custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("marketplace-group", func(t *testing.T) {
+		t.Run("Cluster", MarketplaceCluster)
+	})
+}
+
+func defaultCreateTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("could not get namespace: %v", err)
+	}
+
+	testOperatorSource := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "global-operators",
+			Namespace: namespace,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: "operators",
+		},
+	}
+
+	catalogSourceConfigName := "opsrc-global-operators"
+	configMapName := "csc-cm-opsrc-global-operators"
+	catalogSourceName := "csc-cs-opsrc-global-operators"
+
+	// Create the operatorsource to download the manifests.
+	err = f.Client.Create(
+		context.TODO(),
+		testOperatorSource,
+		&framework.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       cleanupTimeout,
+			RetryInterval: cleanupRetryInterval,
+		})
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig.
+	testCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, testCatalogSourceConfig, namespace, catalogSourceConfigName)
+	if err != nil {
+		return err
+	}
+
```

Comment:
> Assert that the all expected package(s) have been downloaded from quay.io?
How do we do this? Do you want to make direct app registry calls to get the blobs and introspect the blobs and ensure they match? I would suggest against this as that code could potentially have bugs too. We could take the approach of having a separate appregistry namespace for e2e tests with a known set of operator repositories.

Reply:
No, by inspecting the `Status.Packages` field of the `OperatorSource` object. ( My current PR adds this ), or inspecting the `CatalogSourceConfig` object  (Spec.Packages).  Yeah, I assumed it would be done in separate PR(s).

Reply:
Sounds good. Let us restrict comments in this PR targeted towards basic functionality so that we can get this into the CI pipeline soon.